src/models/data/data-set.ts

```diff
@@ -620,15 +636,36 @@ export const DataSet = types.model("DataSet", {
         });
       },
 
-      selectAll(select = true) {
+      selectAllAttributes(select = true) {
+        if (select) {
+          clearAllSelections();
+          self.attributes.forEach(attribute => self.attributeSelection.add(attribute.id));
+        } else {
+          self.attributeSelection.clear();
+        }
```

Comment:
```suggestion
        self.caseSelection.clear();
        if (select) {
          self.attributes.forEach(attribute => self.attributeSelection.add(attribute.id));
        } else {
          self.attributeSelection.clear();
        }
```
No need to clear the attributes before selecting them again.